### PR TITLE
`StrimziKafkaCluster` builder

### PR DESCRIPTION
**Pull request by see-quick**
This PR adds flexibility to `StrimziKafkaCluster` by adapting the Builder pattern. With such change we will easily extend such class adding more parameters. Solves #84.

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
I think the builder pattern is useful. But by removing the original constructors, you are breaking the compatibility. Is that wise / intended? Should we keep them in some way?

**Comment by see-quick** [comment]
> I think the builder pattern is useful. But by removing the original constructors, you are breaking the compatibility. Is that wise / intended? Should we keep them in some way?

Hmmm, I was thinking about it but adding more and more constructors would end up with a lot of constructors. Maybe a less aggressive way would be having both (but `@Deprecate` constructors)?

**Comment by scholzj** [comment]
I think deprecating them would be fine. The question is if you can combine both the deprecated constructors and the builders to allow users to transition smoothly.

**Comment by see-quick** [comment]
> I think deprecating them would be fine. The question is if you can combine both the deprecated constructors and the builders to allow users to transition smoothly.

Yeah, it would be okay I think. It would be a matter of extracting the same code into methods so we don't end up with repetitive code. I will update this PR ten :)

File: src/main/java/io/strimzi/test/container/StrimziKafkaCluster.java [hunk 1]
```diff
@@ -147,10 +126,153 @@ public StrimziKafkaCluster(final int brokersNum, final ToxiproxyContainer proxyC
...
+            .withNetwork(this.network);
+
+        if (this.proxyContainer != null) {
+            this.proxyContainer.setNetwork(this.network);
+        }
+
+        startKafkaCluster(this.additionalKafkaConfiguration);
+    }
+
+    private void startKafkaCluster(final Map<String, String> additionalKafkaConfiguration) {
```
Comment (scholzj) [inline src/main/java/io/strimzi/test/container/StrimziKafkaCluster.java:155]: Does this actually start it? doesn't the `start()` method do it? Should it be maybe called `prepareKafkaCluster` or something?

Comment (see-quick) [reply src/main/java/io/strimzi/test/container/StrimziKafkaCluster.java:155]: Oh yea 🤦 ...`prepareKafkaCluster` sounds better. Thanks

**Review by scholzj — APPROVED** [review]
Thanks.

**Pull request merged**